cupy/linalg/einsum.py

```diff
@@ -315,6 +315,18 @@ def _get_out_shape(shape0, sub0, shape1, sub1, sub_out):
     return out_shape
 
 
+def _reshape_transpose(arr, sub, sub_out):
```

Comment:
Please document the behavior of this function as well as inputs/outputs.

Reply:
I will add docs for this function.